src/runner/index.js

```diff
@@ -67,7 +68,10 @@ export default class Runner extends EventEmitter {
 
     // Run task
     async _getTaskResult (task, browserSet, reporter, testedApp) {
-        task.on('browser-job-done', job => browserSet.releaseConnection(job.browserConnection));
+        task.on('browser-job-done', job => {
+            job.browserConnection.reportTaskStatus(TASK_STATUS.done, { total: job.total, passed: job.passed });
```

Comment:
In this context it's not task status, but browser job status. Browser job can be  done at this point, but whole task is still running. Therefore, maybe it makes sense to rename it to `reportBrowserJobStatus` throughout the codebase?